modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/io/Sink.scala

```diff
@@ -40,4 +48,79 @@ object Sink {
       }
     }
 
+  /** Opens a new file when the existing file exceeds `maxBytes`
+   *  Each file has an integer suffix e.g. /path/to/good.0001
+   */
+  def rotatingFileSink[F[_]: Concurrent: ContextShift](
+    path: Path,
+    maxBytes: Long,
+    blocker: Blocker
+  ): Resource[F, ByteSink[F]] =
+    for {
+      hs <- Hotswap.create[F, OpenState]
+      ref <- Resource.eval(Ref.of[F, RotatingFileState](ClosedState(1)))
```

Comment:
Can't we start directly with an `OpenState` (that would then become `FileState`) to get rid of this resource and have only the `HotSwap` ?